packages/app/components/stackbox/ConfirmStackModal.tsx

```diff
@@ -13,30 +13,76 @@ import {
   ModalBaseProps,
 } from "@/ui";
 import { FromToStackTokenPair } from "@/components/FromToStackTokenPair";
+import { Token } from "@/models/token";
+import { format } from "date-fns";
+import { FREQUENCY_OPTIONS } from "@/models/stack";
 
-export const ConfirmStackModal = ({ isOpen, closeAction }: ModalBaseProps) => {
+interface ConfirmStackModalProps extends ModalBaseProps {
+  fromToken: Token;
+  toToken: Token;
+  amount: string;
+  frequency: FREQUENCY_OPTIONS;
+  startTime: Date;
+  endTime: Date;
+}
+
+const frequencyTitles = {
+  [FREQUENCY_OPTIONS.hour]: "hour",
+  [FREQUENCY_OPTIONS.day]: "day",
+  [FREQUENCY_OPTIONS.week]: "week",
+  [FREQUENCY_OPTIONS.month]: "month",
+};
+
+const HOUR_IN_MILISECONDS = 60 * 60 * 1000;
+const DAY_IN_MILISECONDS = 24 * HOUR_IN_MILISECONDS;
+
+const frequencySeconds = {
+  [FREQUENCY_OPTIONS.hour]: HOUR_IN_MILISECONDS,
+  [FREQUENCY_OPTIONS.day]: DAY_IN_MILISECONDS,
+  [FREQUENCY_OPTIONS.week]: 7 * DAY_IN_MILISECONDS,
+  [FREQUENCY_OPTIONS.month]: 30 * DAY_IN_MILISECONDS,
+};
+
+enum CREATE_STACK_STEPS {
+  approve = "approve",
+  create = "create",
+}
+
+const INITAL_ORDER = 1;
+
+export const ConfirmStackModal = ({
+  fromToken,
+  toToken,
+  amount,
+  frequency,
+  startTime,
+  endTime,
+  isOpen,
+  closeAction,
+}: ConfirmStackModalProps) => {
   const focusBtnRef = useRef<HTMLButtonElement>(null);
+  const [step, setStep] = useState(CREATE_STACK_STEPS.approve);
+
   const [isTransactionLoadingDialogOpen, setTransactionLoadingDialogOpen] =
     useState(false);
 
-  function stack() {
+  const createStack = () => {
     setTransactionLoadingDialogOpen(true);
-  }
-
-  const fromToken = {
-    address: "0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83",
-    symbol: "USDC",
-    decimals: 18,
-    name: "usdc",
   };
 
-  const toToken = {
-    address: "0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1",
-    symbol: "WETH",
-    decimals: 18,
-    name: "wrapped eth",
+  const approveToken = () => {
+    setStep(CREATE_STACK_STEPS.create);
   };
 
+  const estimatedNumberOfOrders =
+    Math.floor(
+      (endTime.getTime() - startTime.getTime()) / frequencySeconds[frequency]
+    ) + INITAL_ORDER;
+
+  const amountPerOrder = (parseFloat(amount) / estimatedNumberOfOrders).toFixed(
+    2
+  );
```

Comment:
maybe this belongs in stack model file

Reply:
I like to have this things declared next to where they are being used. If this was being used elsewhere I definitely would declare under stack model file. What do you think?

Reply:
Makes sense.

Close when it's only used there, but move it to a model when used in more places.